solutions/cf-1216C.mdx

```diff
@@ -0,0 +1,240 @@
+```java
+import java.util.*;
+import java.io.*;
+
+public class Main {
+ 	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio();
+		Rect a = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+		Rect b = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+		Rect c = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+		a = cut(a, b);
+		a = cut(a, c);
+		if(a.x2 - a.x1 == 0 || a.y2 - a.y1 ==0) //area!=0
+			System.out.println("NO");
+		else
+			System.out.println("YES");
+ 		io.close();
+  	}
+	
+	static class Kattio extends PrintWriter {
+		private BufferedReader r;
+		private StringTokenizer st;
+		// standard input
+		public Kattio() { this(System.in, System.out); }
+		public Kattio(InputStream i, OutputStream o) {
+			super(o);
+			r = new BufferedReader(new InputStreamReader(i));
+		}
+		// USACO-style file input
+		public Kattio(String problemName) throws IOException {
+			super(problemName + ".out");
+			r = new BufferedReader(new FileReader(problemName + ".in"));
+		}
+		// returns null if no more input
+		public String next() {
+			try {
+				while (st == null || !st.hasMoreTokens())
+					st = new StringTokenizer(r.readLine());
+				return st.nextToken();
+			} catch (Exception e) { }
+			return null;
+		}
+		public int nextInt() { return Integer.parseInt(next()); }
+		public double nextDouble() { return Double.parseDouble(next()); }
+		public long nextLong() { return Long.parseLong(next()); }
+	}
+	
+	static class Rect{
+		int x1, y1, x2, y2;
+		public Rect(int a, int b, int c, int d){
+			x1 = a;
+			y1 = b;
+			x2 = c;
+			y2 = d;
+		}
+		int area()
+		{
+			return (x2 - x1) * (y2 - y1);
+		}
+	}
+	static Rect cut(Rect a, Rect b) {
+		//if B intersects entirely in the 
+		//x or y direction, cut it
+		if(a.x1 >= b.x1 && b.x2 >= a.x1 && b.y1 <= a.y1 && b.y2 >= a.y2)
+		{
+			//if B cuts A from the left side
+			a.x1 = b.x2;
+			a.x2 = Math.max(a.x2, b.x2); // if b covers A entirely
+		}
+		if(a.x2 >= b.x1 && b.x2 >= a.x2 && b.y1 <= a.y1 && b.y2 >= a.y2)
+		{
+			//if B cuts A from the right side
+			a.x2 = b.x1;
+			a.x1 = Math.min(a.x1, b.x1); // if b covers A entirely
+		}
+		if(a.y1 >= b.y1 && b.y2 >= a.y1 && b.x1 <= a.x1 && b.x2 >= a.x2)
+		{
+			//if B cuts A from the bottom side
+			a.y1 = b.y2;
+			a.y2 = Math.max(a.y2, b.y2); // if b covers A entirely
+		}
+		if(a.y2 >= b.y1 && b.y2 >= a.y2 && b.x1 <= a.x1 && b.x2 >= a.x2)
+		{
+			//if B cuts A from the top side
+			a.y2 = b.y1;
+			a.y1 = Math.min(a.y1, b.y1); // if b covers A entirely
+		}
+		return a;
+	}
+}
+```
+
+</JavaSection>
+
+</LanguageSection>
+
+## Extra Advice
+Especially with rectangle geometry, make a good habit of drawing patterns and observations on a scratch sheet of paper or whiteboard. It helps a lot - David Zhang.
+
+Also if you don't understand the explanation, draw each possibility (Top, Bottom, Left and Right) on a sheet of paper and try to see the connections.
```

Comment:
i love this section
could you possibly put it @ the top in an info block?

Reply:
np, I'll try to do it for future contributions as well